QUARKUS-2489.md

```diff
@@ -0,0 +1,53 @@
+# QUARKUS-2489 OIDC Back channel logout
+
+Jira: https://issues.redhat.com/browse/QUARKUS-2489
+
+This feature is talking about logout flow on an OIDC provider as Keycloak
+
+## Scope of the testing
+
+Backchannel logout applies to `quarkus-oidc` and basically is a standard way to do logout in to a single sign-on service.
+We will use Keycloak as an OIDC provider and the flavor of the extensions (classic or reactive) is not relevant for this coverage. 
+
+Absolute Back-Channel Logout URL is calculated by adding `quarkus.oidc.back-channel-logout.path` to the current endpoint URL and then based on your configuration you could do a local and / or a global logout.
+
+We will create a valid token by an authorization flow and then after validating the token against some authenticated API we will proceed with the logout flow and verify the following constraints:
```

Comment:
Is the following test coverage really necessary?

Aren't you testing Keycloak functionality instead of testing of the integration?

Reply:
I think this comes from specs chapter `2.6.  Logout Token Validation` right :-) So there are 2 situations we need to test:

- Back channel logout goes to Quarkus app and the app sends it to the Keycloak, in such case the response is always going to be empty (see `io.quarkus.oidc.runtime.BackChannelLogoutHandler.RouteHandler`), so just validate response code.
- User logged out in different app (it can be Quarkus app too), and Keycloak informed the tested Quarkus app that logging has happened. Here is where actual value of our tests lay as upstream is not testing this AFAICT. If we don't test it, we can't tell if we are logged out based on Keycloak call back are Quarkus performed local logout. In this case, there is no logout token.

So IMHO no place for logout token validation.

Reply:
Except to what you suggest below - that if we send Quarkus app invalid logout token, we get 400.

Reply:
Although both Keycloak and us calls same URL that keeps back-channel log out token, so there should be a little difference to who calls. IMHO we should test full flow anyway.

Reply:
just to be clear @michalvavrik , you vote for full coverage described here?

Reply:
no @rsvoboda I think it doesn't make sense. We should call back-channel logout endpoint at least once with invalid token so we do that Sergey implemented checks correctly (according to OIDC specs), but I'd not validate logout token myself as that's not our job.

Reply:
by ^^^ I tried to point out that I'm missing in TP bound that we will test callback from KC.